reflections/shape/shape.js

```diff
@@ -813,18 +813,24 @@ shapeReflections = {
 		return target;
 	},
 	assignDeepMap: function(target, source) {
-
+		
 		var hasOwnKey = fastHasOwnKey(target);
 		var getKeyValue = target[getKeyValueSymbol] || shiftedGetKeyValue;
 		var setKeyValue = target[setKeyValueSymbol] || shiftedSetKeyValue;
 
 		shapeReflections.eachKey(source, function(newVal, key){
 			if(!hasOwnKey(key)) {
-				// set no matter what
-				getSetReflections.setKeyValue(target, key, newVal);
+				// Plain objects needs to be serialized to make sure 
+				// newVal is copied not referenced #150
+				if (typeReflections.isMapLike(newVal) && !typeReflections.isObservableLike(newVal)) {
```

Comment:
This won't solve the original issue since it was using a DefineMap. Is that being handled somewhere else?